SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -105,11 +105,44 @@ void USpatialReceiver::LeaveCriticalSection()
 		{
 			OnEntityAddedDelegate.Broadcast(PendingAddEntity);
 		}
+		PendingAddComponents.RemoveAll([PendingAddEntity](const PendingAddComponentWrapper& Component) {return Component.EntityId == PendingAddEntity;});
 	}
 
 	for (Worker_AuthorityChangeOp& PendingAuthorityChange : PendingAuthorityChanges)
 	{
-		HandleActorAuthority(PendingAuthorityChange);
+		if (PendingAuthorityChange.authority != WORKER_AUTHORITY_AUTHORITATIVE)
+		{
+			HandleActorAuthority(PendingAuthorityChange);
+		}
+	}
+
+	for (PendingAddComponentWrapper& PendingAddComponent : PendingAddComponents)
+	{
+		if (StaticComponentView->GetComponentData<UnrealMetadata>(PendingAddComponent.EntityId) == nullptr)
+		{
+			continue;
+		}
+		if (ClassInfoManager->IsGeneratedQBIMarkerComponent(PendingAddComponent.ComponentId))
+		{
+			continue;
+		}
+		if (StaticComponentView->HasAuthority(PendingAddComponent.EntityId, PendingAddComponent.ComponentId))
+		{
+			continue; // Hacky, allows servers to change state if they are going to be authoritative, without us overwriting it with old data
```

Comment:
This would also mean we're dropping the last updates from the runtime right? We sure we want to do this. Have you seen problem otherwise?

Reply:
Yes, otherwise there is a problem. Seems that PlayerState in particular is susceptible to this, as I think the initial data doesn't have the owner or something like that set. Then it's changed locally, and the AddComponent from runtime back would overwrite the owner back to nullptr when it was resolved okay locally.

Reply:
This would break any type of soft handover we might want to implement also. I'm a little concerned about this change - we should probably create a follow up ticket and track it here, as this is just as likely to sting us as it is to help.

Reply:
Ah, yes, definitely, this is a hacky workaround for current behaviour.
I will create a ticket.